app/src/org/commcare/fragments/connectMessaging/ConnectMessageChannelConsentBottomSheet.java

```diff
@@ -33,6 +33,15 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         ConnectMessagingChannelRecord channel = ConnectMessagingDatabaseHelper.getMessagingChannel(requireContext(),
                 args.getChannelId());
 
+
+
+        if (channel == null) {
+            // Channel not found, show error and pop back
+            Toast.makeText(requireContext(), getString(R.string.connect_messaging_channel_not_found), Toast.LENGTH_LONG).show();
+            NavHostFragment.findNavController(this).popBackStack();
+            return binding.getRoot();
+        }
```

Comment:
this seems to signify a coding error since we passed an argument that is not a channel, I would much prefer just adding a `Objects.requireNotNull(channel)` call here and let it crash as a consequence with clear error.